### DISCUSSION TITLE
Size of the view window for park entrance depends on which tab was viewed previously.

### DISCUSSION THREAD

**Discussion by ocalhoun6** [OpenRCT2/OpenRCT2#15730] [Feature requests]
OS: Windows 8
Version: e.g. 0.2.
Commit/Build: 2350cd1

The size of the view window for the park entrance window depends on which tab of that window you had open previously.

For example:
1: Open scenario goal tab:
![turtle cove 2019-01-19 23-30-27](https://user-images.githubusercontent.com/26573303/51437415-e700dc00-1c52-11e9-85c1-04124524ff2a.png)
(It's a bit bigger than it needs to be for this much text, but I assume it's made that big so it can fit any possible text?)
2: Switch to viewport tab:
![turtle cove 2019-01-19 23-30-30](https://user-images.githubusercontent.com/26573303/51437419-039d1400-1c53-11e9-9339-9dd8a33f2eae.png)
It has the same dimensions as the scenario goal tab, making it taller than usual.
3: Switch to the park rating tab:
![turtle cove 2019-01-19 23-30-23](https://user-images.githubusercontent.com/26573303/51437423-1b749800-1c53-11e9-84db-abb327ef211d.png)
The window goes back to the normal size.
4: Switch back to the viewport tab:
![turtle cove 2019-01-19 23-30-25](https://user-images.githubusercontent.com/26573303/51437427-292a1d80-1c53-11e9-847c-9a8022c540d2.png)
It now has the normal size and aspect ratio again.

**Comment by ghost** [comment]
I don't think that's a bug since the viewport is fitting the window size as a camera inlaid the window panel.
Also the entrance viewport can be resized at the right bottom of the same window.

Checking if it's also happening on rct2 classic could be great

**Comment by Broxzier** [comment]
Since the main window is resizable, I don't think this is a bug either. It simply uses the size that it already has.

**Comment by IntelOrca** [comment]
The only alteration I would consider making is recording the previous size the window was last time the tab was active but I don't know if everyone would like that.

**Comment by ocalhoun6** [comment]
Yes, the window is resizable, but let's see what happens when I set a custom size to the window...

Custom size set:
![forest frontiers 2019-01-20 12-16-49](https://user-images.githubusercontent.com/26573303/51444547-b8641f00-1cad-11e9-81c6-9f7a43e1a8f1.png)

Switch to _any_ different tab:
![forest frontiers 2019-01-20 12-16-53](https://user-images.githubusercontent.com/26573303/51444550-c44fe100-1cad-11e9-8a07-0c12b2f8698b.png)

The view window's size is now the size of the previous tab:
![forest frontiers 2019-01-20 12-16-56](https://user-images.githubusercontent.com/26573303/51444554-ce71df80-1cad-11e9-9672-321d33b9f335.png)

Personally, I think the viewport tab should ether have its own default size, which it always goes back to when you switch to its tab (after all, every other tab behaves this way) ... or it should behave like IntelOrca said, remembering the size it had last time it was open. (Which is probably a bit more difficult to code, but would be nicer.) Either one would prevent the seemingly odd behavior in the first post. (As long as the size it remembers is initialized as the default 'normal' size -- it's possible to open the window on a different tab, switch to scenario goals, then switch to the viewport tab for the first time.)

If anyone _is_ going to work on this, it's probably worth looking into the same thing on other types of viewport windows: peep, ride, staff, etc. Especially if you were going to implement IntelOrca's idea of how to do it.
